v23.1/advanced-changefeed-configuration.md

```diff
@@ -9,37 +9,140 @@ docs_area: stream_data
-- Use the [`compression` option](create-changefeed.html#compression-opt) in cloud storage sinks with JSON to compress the changefeed data files.
-- Increase the [`changefeed.backfill.concurrent_scan_requests` setting](cluster-settings.html), which controls the number of concurrent scan requests per node issued during a backfill event. The default behavior, when this setting is at `0`, is that the number of scan requests will be 3 times the number of nodes in the cluster (to a maximum of 100). While increasing this number will allow for higher throughput, it **will increase the cluster load overall**, including CPU and IO usage.
--  Enable the [`kv.rangefeed.catchup_scan_iterator_optimization.enabled` setting](cluster-settings.html) to have [rangefeeds](create-and-configure-changefeeds.html#enable-rangefeeds) use time-bound iterators for catch-up scans when possible. Catch-up scans are run for each rangefeed request. This setting improves the performance of changefeeds during some [range-split operations](architecture/distribution-layer.html#range-splits).
+
+#### Compression
+
+- Use the [`compression` option](create-changefeed.html#compression-opt) when you create a changefeed emitting data files to a [cloud storage sink](changefeed-sinks.html#cloud-storage-sink). For larger files, set `compression` to the `zstd` format.
+- Use the `snappy` compression format to emit messages to a [Kafka](changefeed-sinks.html#kafka-compression) sink. If you're intending to do large batching for Kafka, use the `lz4` compression format.
+
+#### File size
+
+To configure changefeeds emitting to [cloud storage sinks](changefeed-sinks.html#cloud-storage-sink) for high throughput, you should consider:
+
+- Increasing the [`file_size`](create-changefeed.html#file-size) parameter to control the size of the files that the changefeed sends to the sink. The default is `16MB`. To configure for high throughput, we recommend `32MB`–`128MB`. Note that this is not a hard limit, and a changefeed will flush the file when it reaches the specified size.
+- When you [compress](#compression) a file, it will contain many more events.
+- File size is also dependent on what kind of data the changefeed job is writing. For example, large JSON blobs will quickly fill up the `file_size` value compared to small rows.
+- When you change or increase `file_size`, ensure that you adjust the `changefeed.memory.per_changefeed_limit` [cluster setting](cluster-settings.html), which has a default of `512MiB`. Buffering messages can quickly reach this limit if you have increased the file size.
+
+### Configuring for tables with many ranges
+
+If you have a table with 10,000 or more [ranges](architecture/overview.html#range), you should consider increasing the following two [cluster settings](cluster-settings.html). We strongly recommend increasing these settings slowly. That is, increase the setting and then [monitor](monitor-and-debug-changefeeds.html) its impact before adjusting further:
```

Comment:
@miretskiy I took this passage from the google docs notes you and abbey made, let me know if I messed up the meaning here at all.

Reply:
this is good